kong/plugins/zipkin/opentracing.lua

```diff
@@ -7,15 +7,15 @@ A plugin that derives this should:
   - Implement a :initialise_request(conf, ctx) method if it needs to do per-request initialisation
 ]]
 
-local BasePlugin = require "kong.plugins.base_plugin"
 local subsystem = ngx.config.subsystem
 
-local OpenTracingHandler = BasePlugin:extend()
-OpenTracingHandler.VERSION = "scm"
+local OpenTracingHandler = {
+  VERSION = "scm",
+  -- We want to run first so that timestamps taken are at start of the phase
+  -- also so that other plugins might be able to use our structures
+  PRIORITY = 100000,
+}
 
--- We want to run first so that timestamps taken are at start of the phase
--- also so that other plugins might be able to use our structures
-OpenTracingHandler.PRIORITY = 100000
 
 function OpenTracingHandler:new(name)
   OpenTracingHandler.super.new(self, name or "opentracing")
```

Comment:
`super` is being set to `self` below... Wouldn't this cause an infinite loop if this line was ever to run? (Seems to indicate `:new` is never being called, in which case `self.conf_to_tracer` needs to be initialized somewhere else)

Reply:
> infinite loop

Removed `:new`, so that won't happen here.

> `self.conf_to_tracer`

That member variable seemed to be a simple "cache" for traces. I made it a "module local cache" instead.